homeassistant/components/motionmount/number.py

```diff
@@ -0,0 +1,75 @@
+"""Support for MotionMount numeric control."""
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MotionMountCoordinator
+from .entity import MotionMountEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Vogel's MotionMount from a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    unique_id = format_mac(coordinator.mm.mac.hex())
+
+    async_add_entities(
+        [
+            MotionMountExtension(coordinator, unique_id),
+            MotionMountTurn(coordinator, unique_id),
+        ]
+    )
+
+
+class MotionMountExtension(MotionMountEntity, NumberEntity):
+    """The target extension position of a MotionMount."""
+
+    _attr_native_max_value = 100
+    _attr_native_min_value = 0
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_translation_key = "motionmount_extension"
+
+    def __init__(self, coordinator: MotionMountCoordinator, unique_id: str) -> None:
+        """Initialize Extension number."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{unique_id}-extension"
+        self._attr_native_value = self.coordinator.data["extension"]
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self._attr_native_value = self.coordinator.data["extension"]
+        self.async_write_ha_state()
```

Comment:
```suggestion

    @property
    def native_value(self) -> float:
        return self.coordinator.data["extension"]
```
What do you think of this setup? This way you keep the logic to get the native value at one place and don't duplicate it. Other solution would be to extract the setting of `_attr_*` in a separate function. Checkout Airzone (or cloud, don't know out of the top of my head)